src/overlays/actors/ovl_Bg_Kin2_Bombwall/z_bg_kin2_bombwall.c

```diff
@@ -28,50 +34,191 @@ const ActorInit Bg_Kin2_Bombwall_InitVars = {
     (ActorFunc)BgKin2Bombwall_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80B6E6F0 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000008, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 60, 60, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B6E748[] = {
+s32 BgKin2Bombwall_IsHitFromNearby(BgKin2Bombwall* this, GlobalContext* globalCtx) {
+    Actor* bombwallCollider;
+
+    if (this->collider.base.acFlags & AC_HIT) {
+        bombwallCollider = this->collider.base.ac;
+        // Distance check required to only react to sufficiently close explosions.
+        if ((bombwallCollider != NULL) &&
+            (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &bombwallCollider->world.pos) < 6400.0f)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+static Color_RGBA8 sPrimColor = { 210, 210, 210, 255 };
+static Color_RGBA8 sEnvColor = { 140, 140, 140, 255 };
+static Vec3f sDustAccel = { 0.0f, 0.33f, 0.0f };
+static s8 sRandomYOffsets[] = { -60, -34, -8, 18, 44 };
+static s16 sScales[] = { 25, 23, 21, 19, 17, 15, 13, 10 }; // Scales for random explosion debris.
+
+void BgKin2Bombwall_SpawnEffects(BgKin2Bombwall* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f spD8;
+    Vec3f spCC;
+    s32 j;
+    s32 k;
+    f32 temp_a0;
+    s32 pad;
+    s16 phi_s0;
+    s16 phi_s1;
+
+    Matrix_RotateYS(this->dyna.actor.shape.rot.y, 0);
+
+    for (i = 0, k = 0; i < 6; i++) {
+        temp_a0 = (i + 1) * 15.f;
+        for (j = 0; j < ARRAY_COUNT(sRandomYOffsets); j++) {
+            k++;
+            k &= 7;
+
+            spD8.x = sRandomYOffsets[j] + (s32)(((u32)Rand_Next()) >> 0x1C);
```

Comment:
The u32 cast is not required here now.
```suggestion
            spD8.x = sRandomYOffsets[j] + (s32)(Rand_Next() >> 0x1C);
```
Is the s32 cast needed?